web/app/src/ui/Layout.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {Suspense, useEffect} from "react";
+import {create} from "zustand";
+import {Outlet} from "react-router-dom";
+import {Menu} from "./Menu";
+import {LoaderView} from "./kit/Loader";
+
+/**
+ * Layout data store.
+ */
+const useLayoutStore = create<{
+	title: string;
+}>(() => ({
+	title: "",
+}));
+
+/**
+ * Layout component.
+ */
+export function Layout({children}: React.PropsWithChildren<{}>) {
+	const title = useLayoutStore(state => state.title);
+
+	return (
+		<>
+			<header>
+				<h1>{title}</h1>
+			</header>
```

Comment:
This could be outside of the layout and directly inside pages. That way, you wouldn't need any special system to communicate the title up.